File: openweathermap-connector-scala/src/com/openweathermap/kafka/connect/WeatherAPIConfig.scala
package com.openweathermap.kafka.connect

import java.util.Map
import org.apache.kafka.common.config.AbstractConfig
import org.apache.kafka.common.config.ConfigDef
import WeatherAPIConfig._
import scala.beans.{BeanProperty, BooleanBeanProperty}
import scala.collection.JavaConversions._

object WeatherAPIConfig {

  val OPEN_WEATHER_API_KEY: String = "open.weather.api.key"
  private val OPEN_WEATHER_API_KEY_DOC: String = "The API key used by connector to poll data"

  val CITIES: String = "cities"
  private val CITIES_DOC: String = "Cities that you want to check weather for"

  val POLL_FREQUENCY: String = "poll.frequency"
  private val POLL_FREQUENCY_DOC: String = "Duration between each request to Weather API"

  val KAFKA_TOPIC: String = "kafka.topic"
  private val KAFKA_TOPIC_DOC: String = "Kafka Topic to send data to"

  def config(): ConfigDef =
    new ConfigDef()
      .define(OPEN_WEATHER_API_KEY,
              ConfigDef.Type.STRING,
              ConfigDef.Importance.HIGH,
              OPEN_WEATHER_API_KEY_DOC)
      .define(CITIES,
              ConfigDef.Type.STRING,
              ConfigDef.Importance.HIGH,
              CITIES_DOC)
      .define(POLL_FREQUENCY,
              ConfigDef.Type.LONG,
              10000L,
              ConfigDef.Importance.MEDIUM,
              POLL_FREQUENCY_DOC)
      .define(KAFKA_TOPIC,
              ConfigDef.Type.STRING,
              ConfigDef.Importance.HIGH,
              KAFKA_TOPIC_DOC)

}

class WeatherAPIConfig(originals: Map[String, _])
    extends AbstractConfig(config(), originals) {

  @BeanProperty
  val openWeatherApiKey: String = this.getString(OPEN_WEATHER_API_KEY)

  @BeanProperty
  val cities: String = this.getString(CITIES)

  @BeanProperty
  val pollFrequency: java.lang.Long = this.getLong(POLL_FREQUENCY)

  @BeanProperty
  val kafkaTopic: String = this.getString(KAFKA_TOPIC)

}

File: openweathermap-connector-java/src/com/openweathermap/kafka/connect/WeatherAPIConfig.java
package com.openweathermap.kafka.connect;

import java.util.Map;

import org.apache.kafka.common.config.AbstractConfig;
import org.apache.kafka.common.config.ConfigDef;

public class WeatherAPIConfig extends AbstractConfig {

	public static final String OPEN_WEATHER_API_KEY = "open.weather.api.key";
    private static final String OPEN_WEATHER_API_KEY_DOC = "The API key used by connector to poll data";

    public static final String CITIES = "cities";
    private static final String CITIES_DOC = "Cities that you want to check weather for";

    public static final String POLL_FREQUENCY = "poll.frequency";
    private static final String POLL_FREQUENCY_DOC = "Duration between each request to Weather API";

    public static final String KAFKA_TOPIC = "kafka.topic";
    private static final String KAFKA_TOPIC_DOC = "Kafka Topic to send data to";

    private final String openWeatherApiKey;
    private final String cities;
    private final Long pollFrequency;
    private final String kafkaTopic;

    public WeatherAPIConfig(Map<String, ?> originals) {
        super(config(), originals);
        this.openWeatherApiKey = this.getString(OPEN_WEATHER_API_KEY);
        this.cities = this.getString(CITIES);
        this.pollFrequency = this.getLong(POLL_FREQUENCY);
        this.kafkaTopic = this.getString(KAFKA_TOPIC);
    }

    public static ConfigDef config() {
        return new ConfigDef()
                .define(OPEN_WEATHER_API_KEY, ConfigDef.Type.STRING,
                        ConfigDef.Importance.HIGH, OPEN_WEATHER_API_KEY_DOC)
                .define(CITIES, ConfigDef.Type.STRING,
                        ConfigDef.Importance.HIGH, CITIES_DOC)
                .define(POLL_FREQUENCY, ConfigDef.Type.LONG, 10000L,
                        ConfigDef.Importance.MEDIUM, POLL_FREQUENCY_DOC)
                .define(KAFKA_TOPIC, ConfigDef.Type.STRING,
                        ConfigDef.Importance.HIGH, KAFKA_TOPIC_DOC);
    }

    public String getOpenWeatherApiKey() {
        return openWeatherApiKey;
    }

    public String getCities() {
        return cities;
    }

    public Long getPollFrequency() {
        return pollFrequency;
    }

    public String getKafkaTopic() {
        return kafkaTopic;
    }

	
}